src/components/Foods.js

```diff
@@ -0,0 +1,122 @@
+import React from 'react';
+import foodSeed from '../foods.json'
+import FoodBox from './FoodBox'
+import AddFood from './AddFood'
+import TodayFood from './TodayFood'
+
+
+class Foods extends React.Component {
+    constructor(props){
+        super(props)
+        this.state ={
+            foodList: foodSeed,
+            filteredFoods: foodSeed,
+            formShow: false,
+            TodayFoods: []
+        }
+    }
+
+    formAppear = () => {
+        this.setState({
+            formShow: true
+        })
+    }
+
+    handleAdd = (e) => {
+        e.preventDefault()
+        let foodName        = e.currentTarget.foodName.value
+        let foodCalories    = e.currentTarget.foodCalories.value
+        let foodImage       = e.currentTarget.foodImage.value
+        let foodQuantity    = e.currentTarget.foodQuantity.value
+        let newFood = [
+            {name:      foodName, 
+            calories:   foodCalories,
+            image:      foodImage,
+            quantity:   foodQuantity   
+            }, ...this.state.foodList]
+
+        this.setState({
+          foodList:       newFood,
+          filteredFoods:  newFood,
+          formShow:       false
+        })
+    }
+
+    handleSearch = (e) => { 
+      let searchName = e.currentTarget.value
+      let cloneFoods = this.state.foodList.filter((food) => {
+        return food.name.startsWith(searchName.toUpperCase())
```

Comment:
Currently it only works for one letter, after the second, it breaks. Try using the toUpperCase on the food.name as well. Also, it would be best to use .includes instead of .startsWith as the first one will check for all appearances. So you could try to find "roast" and get a result :)